runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -193,7 +193,7 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
         for (int x = 0; x < rt.getParameters().getTrimRetry(); x++) {
             // Linearize this read against a timestamp
             final long timestamp = rt.getSequencerView()
-                            .query(getStreamID()).getToken().getSequence();
+                            .query(getStreamID()).getStreamTails().get(0);
```

Comment:
Maybe introduce `SequencerView::query(UUID streamId)`, then the external consumers don't have to access the list directly.

Reply:
You mean for the return type to be a long? ... for cases like the checkpointer we'd still need to go through the other API cause we want the global tail as well.